src/charm.py

```diff
@@ -195,15 +214,28 @@ def _on_start(self, event: StartEvent) -> None:
             )
             event.defer()
 
-    def _on_update_status(self, _):
+    def _on_upgrade(self, event) -> None:
+        container = self.unit.get_container(Config.CONTAINER_NAME)
+        try:
+            self._configure_layers(container=container)
+        except ContainerNotReadyError:
+            self.status.set_and_share_status(Config.Status.UNHEALTHY_UPGRADE)
+            logger.error("Failed to replan")
+            event.defer()
+            return
+
+        self.status.set_and_share_status(ActiveStatus())
```

Comment:
should we really set active status? might it be better to use the status setting from k8s_upgrades?

Reply:
I'll do the same as mongodb k8s :)

Reply:
502f904